addons/l10n_sa_edi/models/account_edi_xml_ubl_21_zatca.py

```diff
@@ -256,6 +257,8 @@ def _l10n_sa_get_monetary_vals(self, invoice, vals):
             # - BR-KSA-80: To calculate payable amount, we deduct prepaid amount from total tax inclusive amount
             prepaid_amount = downpayment_vals['total_amount']
             payable_amount = tax_inclusive_amount - prepaid_amount
+            if float_is_zero(payable_amount, precision_rounding=invoice.currency_id.rounding):
+                payable_amount = 0.0
```

Comment:
Why not rounding the payable amount?

something like `payable_amount = invoice.currency_id.round(tax_inclusive_amount - prepaid_amount)`?